### PR TITLE
Add support for customer login

### PR DESCRIPTION
**Pull request by mattolson**
* Add instance method on Customer object to generate
  a login token for that customer. This only works if
  the app has the store_v2_customer_login scope and is
  installed on the store.

### CODE REVIEW

**Comment by mattolson** [comment]
@philipmuir @pedelman

File: lib/bigcommerce/resources/customers/customer.rb [hunk 1]
```diff
@@ -26,5 +28,21 @@ class Customer < Resource
     def self.count(params = {})
       get 'customers/count', params
     end
+
+    # Generate a token that can be used to log the customer into the storefront.
+    # This requires your app to have the store_v2_customers_login scope and to
+    # be installed in the store.
+    def login_token
```
Comment (pedelman) [inline lib/bigcommerce/resources/customers/customer.rb]: In order to protect the thread safety of this, we should allow a user to pass in the store_hash, client_id, and client_secret (AKA a config)

File: examples/customers/customer_login.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+require 'bigcommerce'
+require 'securerandom'
```
Comment (pedelman) [inline examples/customers/customer_login.rb]: dont need secure random here

File: lib/bigcommerce/resources/customers/customer.rb [hunk 3]
```diff
@@ -26,5 +28,21 @@ class Customer < Resource
...
+      payload = {
+        'iss' => Bigcommerce.config.client_id,
+        'iat' => Time.now.to_i,
+        'jti' => SecureRandom.hex(16),
```
Comment (pedelman) [inline lib/bigcommerce/resources/customers/customer.rb]: Not sure this is good enough, maybe client_id and a uuid?

https://self-issued.info/docs/draft-ietf-oauth-json-web-token.html#rfc.section.4.1.7

**Comment by pedelman** [comment]
Mixed opinion about adding a specific JWT gem as a dependency, but its is a very useful thing to have.

What are others thoughts?

**Comment by pedelman** [comment]
A few minor comments, but code looks very close and the tests are awesome. Do you feel this is a patch or minor version bump? I feel this is probably minor version since we are adding a new dependency.

File: lib/bigcommerce/resources/customers/customer.rb (hunk 1 above)
Comment (mattolson) [inline lib/bigcommerce/resources/customers/customer.rb]: Good idea, will do

File: lib/bigcommerce/resources/customers/customer.rb (hunk 3 above)
Comment (mattolson) [inline lib/bigcommerce/resources/customers/customer.rb]: You're right, will change to uuid

**Comment by mattolson** [comment]
@pedelman ♻️

**Comment by mattolson** [comment]
@pedelman Semver speaks to the API of our library. I don't believe that adding a dependency changes that, since it is completely internal to the library. However, since we are adding functionality in a backward compatible way, it would be a minor version bump, i.e. 1.1.0

File: README.md [hunk 4]
```diff
@@ -92,6 +92,16 @@ For more information about configuring SSL with Faraday, please see the followin
...
 - [Faraday: Setting up SSL certificates](https://github.com/lostisland/faraday/wiki/Setting-up-SSL-certificates)
 
+### Customer Login API
+If you want to generate tokens for storefront login using the Customer Login API, you need to configure your app's client secret.
+
+- ```client_secret```: Obtained from the on the BigCommerce [Developer Portal's](http://developer.bigcommerce.com) "My Apps" section.
+
+```rb
+Bigcommerce.configure do |config|
+  config.client_secret = ENV['BC_CLIENT_SECRET']
```
Comment (pedelman) [inline README.md]: Maybe this example can show that you also need a store_hash and a client_id

File: lib/bigcommerce/resources/customers/customer.rb [hunk 5]
```diff
@@ -26,5 +29,21 @@ class Customer < Resource
     def self.count(params = {})
       get 'customers/count', params
     end
+
+    # Generate a token that can be used to log the customer into the storefront.
+    # This requires your app to have the store_v2_customers_login scope and to
+    # be installed in the store.
+    def login_token(config = Bigcommerce.config)
```
Comment (pedelman) [inline lib/bigcommerce/resources/customers/customer.rb]: Thoughts on using keyword args here so the api is similar to the other hash param methods? just change the `=` to `:`

**Comment by mattolson** [comment]
@pedelman ♻️

**Comment by pedelman** [comment]
Looks good to me! :+1:

Lets sync about when we want to cut a 1.1.0 release for this -- I think its okay for it to sit in master since 1.0.0 is already released.

**Comment by mattolson** [comment]
Agreed, no rush to release a new version

**Pull request merged**